### PR TITLE
Remove some major comments

### PR DESCRIPTION
**Pull request by franmomu**

### CODE REVIEW

File: src/Datagrid/ProxyQueryInterface.php [hunk 1]
```diff
@@ -37,20 +37,31 @@ public function execute(array $params = [], ?int $hydrationMode = null);
     /**
      * @param mixed[] $parentAssociationMappings
      * @param mixed[] $fieldMapping
+     *
+     * @return static
      */
-    public function setSortBy(array $parentAssociationMappings, array $fieldMapping): self;
+    public function setSortBy(array $parentAssociationMappings, array $fieldMapping);
```
[the remark this answers is not in the document]
Comment (franmomu) [reply src/Datagrid/ProxyQueryInterface.php:43]: The problem is that for php 7.3, it is not right: https://3v4l.org/fOjGL

[the remark this answers is not in the document]
Comment (VincentLanglet) [reply src/Datagrid/ProxyQueryInterface.php:43]: Yes, that's right I said "After the php 7.3 drop".

Until, you can write
```
/**
 * @return ProxyQuery
 */
public function setSortBy(array $parentAssociationMappings, array $fieldMapping): ProxyQueryInterface
```

In https://github.com/sonata-project/SonataAdminBundle/issues/6404 we took a lot of time adding typehint everywhere we could. I feel bad removing them now, especially when a solution exists.

[the remark this answers is not in the document]
Comment (franmomu) [reply src/Datagrid/ProxyQueryInterface.php:43]: > Yes, that's right I said "After the php 7.3 drop".

You wrote:

> Why not keeping the return typehint and using the return static annotation in the persistence bundle instead ?
> 
> ```
> /**
>  * @return static
>  */
> public function setSortBy(array $parentAssociationMappings, array $fieldMapping): self
> ```
> 
> It should do the work.

That's why I said that it doesn't work.

> Until, you can write
> 
> ```
> /**
>  * @return ProxyQuery
>  */
> public function setSortBy(array $parentAssociationMappings, array $fieldMapping): ProxyQueryInterface
> ```
> 
> In #6404 we took a lot of time adding typehint everywhere we could. I feel bad removing them now, especially when a solution exists.

This code makes phpstan fail: https://phpstan.org/r/b3c2b4dc-277e-48c1-87d9-a3a204a8413f

**Review by VincentLanglet — APPROVED** [review]

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @franmomu !

**Pull request merged**